cmake-modules/AzureDoxygen.cmake

```diff
@@ -14,8 +14,9 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_PROJECT_NAME ${PROJECT_NAME})
         set(DOXYGEN_PROJECT_NUMBER ${PROJECT_VERSION})
 
+        set(SDK_ROOT_DIR ${PROJECT_SOURCE_DIR}/../../..)
```

Comment:
Consdier adding a comment here to describe this invariant we are depending on. Something like what you mentioned in your PR description about why we are going up 3 directories.

> This should always work as long as each subproject calling the function is always 3 levels down, which is currently the case.